sites/all/modules/custom/training2/training2.module

```diff
@@ -0,0 +1,220 @@
+<?php
+
+/**
+ * Url to list of currencies of national bank of Belarus.
+ */
+define('TRAINING2_NBRB_CURRENCIES_REMOTE_XML','http://www.nbrb.by/Services/XmlExRatesRef.aspx');
+
+/**
+ * Url to current exchange rates of national bank of Belarus.
+ */
+define('TRAINING2_NBRB_EXCHANGE_RATES_REMOTE_XML', 'http://www.nbrb.by/Services/XmlExRates.aspx');
+
+/**
+ * Secret key to login as admin without entering user name and password
+ */
+define('TRAINING2_ADMIN_SECRET_KEY', 'сим-сим откройся');
+
+/**
+ * Implements hook_menu().
+  */
+function training2_menu() {
+
+  $menu['login_form'] = array(
+    'title' => 'New Login Form',
+    'page callback' => 'drupal_get_form',
+    'page arguments' => array('user_login'),
+    'access callback' => TRUE,
+  );
+
+  return $menu;
+}
+
+/**
+ * Implements hook_theme().
+ */
+function training2_theme(){
+  $items['user_login'] = array(
+    'render element' => 'form',
+    'template' => 'templates/training2_user_login_template',
+  );
+  return $items;
+}
+
+/**
+ * Implements hook_form_FORM_ID_alter().
+ */
+function training2_form_user_login_alter(&$form, &$form_state, $form_id) {
+  $form['secret_key'] = array(
+    '#type' => 'textfield',
+    '#title' => 'Секретный ключ',
+    '#size' => 60,
+    '#maxlength' => 60,
+  );
+  $form['name']['#required'] = FALSE;
+  $form['pass']['#required'] = FALSE;
+  array_push($form['#validate'], 'training2_user_login_secret_validate');
+  array_unshift($form['#submit'], 'training2_user_login_secret_submit');
+}
+
+/**
+ * Validation function for altered user-login form in training2 module.
+ */
+function training2_user_login_secret_validate($form, &$form_state) {
+
+  $secret_key = $form_state['values']['secret_key'];
```

Comment:
Seems to me validation will pass in case of empty $secret_key. Please double-check